Libs/MRML/Core/vtkMRMLBitStreamVolumeNode.cxx

```diff
@@ -0,0 +1,247 @@
+#include "vtkMRMLBitStreamVolumeNode.h"
+
+// MRML includes
+#include "vtkMRMLScene.h"
+#include "vtkMRMLNRRDStorageNode.h"
+
+// VTK includes
+#include <vtkCollection.h>
+#include <vtkMatrix4x4.h>
+#include <vtkNew.h>
+#include <vtkObjectFactory.h>
+#include <vtkXMLUtilities.h>
+
+//----------------------------------------------------------------------------
+vtkMRMLNodeNewMacro(vtkMRMLBitStreamVolumeNode);
+
+//----------------------------------------------------------------------------
+// vtkMRMLBitStreamVolumeNode methods
+
+//-----------------------------------------------------------------------------
+vtkMRMLBitStreamVolumeNode::vtkMRMLBitStreamVolumeNode()
+{
+  this->CompressionDevice = NULL;
+  this->FrameUpdated = false;
+  this->KeyFrameReceived = false;
```

Comment:
There are 32 potential potential states of  these 5 flags, which is quite difficult to keep track of and probably not all combinations are correct. Would it be possible to use enum(s) to enumerate valid states?

Reply:
I remove one unnecessary flag now we have 4 flags:
FrameUpdated, KeyFrameReceived, KeyFrameDecoded, KeyFrameUpdated.
FrameUpdated and the other three key frame flags are independent. 
And most combination of the key frame flags are valid.  i will keep the implementation as it's now.